### PR TITLE
fix: update error in the tic-tac-toe update logic

### PR DESCRIPTION
**Pull request by Gozala**
There was a little error in the logic, possible caused by unintuitive choices made by database. Specifically when you assert a fact it does not retract previous one even if the [entity, attribute] pair is the same. That is intentionally so because it models all relations as 1:many and never 1:1. Which in turn because schema changes from 1:1 to 1:n tends to be painful and if you just assume 1:n you can avoid that pain down the line.

In other words if you want cardinality 1 you just need to always retract previous fact and assert new fact 😅 Perhaps we'll introduce something like `Swap` in the future for convenience, but right now it's up to the program to retract facts that are no longer true.

P.S.: Sorry for the large delta it's just my editor passes through prettier and I was lazy to undo it.

### CODE REVIEW

File: src/tic-tac-toe.behavior.js [hunk 1]
```diff
@@ -62,18 +60,20 @@ export default System.be({
...
   // Handle player moves
   makeMove: {
-    select: { self: DB.self, currentPlayer, gameStatus, event },
+    select: { self: DB.self, currentPlayer, event },
     where: [
       { Case: [DB.self, 'currentPlayer', currentPlayer] },
-      { Case: [DB.self, 'gameStatus', gameStatus] },
+      { Case: [DB.self, 'gameStatus', 'ongoing'] },
       { Case: [DB.self, `/on/click`, event] },
-      { Is: [gameStatus, 'ongoing'] },
```
Comment (Gozala) [inline src/tic-tac-toe.behavior.js:70]: This change was not strictly necessary, however I suspect that maybe it was on obvious that this simplified version would have worked. In this case query only matches when `gameStatus` is `ongoing` which I assume was the intention here.

File: src/tic-tac-toe.behavior.js [hunk 2]
```diff
@@ -62,18 +60,20 @@ export default System.be({
...
     update: ({ self, currentPlayer, event }) => {
-      const { x, y } = event.value.target.dataset;
+      const { x, y } = event.value.target.dataset
       return [
         { Associate: [self, `${x}/${y}`, currentPlayer] },
-        { Associate: [self, 'currentPlayer', currentPlayer === 'X' ? 'O' : 'X'] }
+        { Disassociate: [self, 'currentPlayer', currentPlayer] },
+        {
+          Associate: [self, 'currentPlayer', currentPlayer === 'X' ? 'O' : 'X'],
+        },
```
Comment (Gozala) [inline src/tic-tac-toe.behavior.js:76]: This is an only thing that required change, and caused subtle bug you've observed. What happened was that you end up with `[self, 'currentPlayer', 'X']` in first run and then with `[self, 'currentPlayer', '0']` in second run. Which meant that `renderBoard` produced two game states one per each.

Rest of the issue was than caused by this hacky code in the render loop

https://github.com/bfollington/co-dom-todomvc/blob/9fa1e167296913536d1c944fd85971aa6b12a91e/src/system.js#L226-L255

It simply queries db to get  value of `[self, '/common/ui', ui]` and then just renders the first match ignoring the fact that there could be several matches, which would be a case because of the behavior described above. I have not verified this, but I suspect that it just happens to be that second game comes out as first match (since order is not deterministic) which is why it appears like game stops reacting after second step 😅

Comment (Gozala) [reply src/tic-tac-toe.behavior.js:76]: We definitely could handle case where there are multiple UIs better than simply picking the first, but I have put all this in few hours so I didn't have time to think through. For what it's worth I think in terms of our OS it's not that bad it just means we have multiple views to render 😅 It will appear like forks of the gameplay

File: src/rogue.behavior.js [hunk 3]
```diff
@@ -59,6 +59,8 @@ export default System.be({
       }
 
       return [
+        { Disassociate: [self, 'playerX', playerX] },
+        { Disassociate: [self, 'playerY', playerY] },
```
Comment (Gozala) [inline src/rogue.behavior.js:63]: Same issue as with the other example, provided detailed explanation in the comment below
